libs/unity/library/Runtime/Scripts/PeerConnection.cs

```diff
@@ -278,26 +271,33 @@ public static Task<IReadOnlyList<VideoCaptureDevice>> GetVideoCaptureDevicesAsyn
         /// that the <see cref="Peer"/> property is non-<c>null</c> to confirm that the connection
         /// is in fact initialized.
         /// </remarks>
-        private Task InitializeAsync(CancellationToken token = default(CancellationToken))
+        private async Task<WebRTC.PeerConnection> InitializeAsync(CancellationToken token)
         {
-            CreateNativePeerConnection();
+            // Create the peer connection managed wrapper and its native implementation
+            var nativePeer = new WebRTC.PeerConnection();
+
+            nativePeer.AudioTrackAdded +=
+                (RemoteAudioTrack track) =>
+                {
+                    // Tracks will be output by AudioReceivers, so avoid outputting them twice.
+                    track.OutputToDevice(false);
+                };
 
             // Ensure Android binding is initialized before accessing the native implementation
             Android.Initialize();
 
+            // Continue the task outside the Unity app context, in order to avoid deadlock
+            // if OnDisable waits on this task.
 #if UNITY_WSA && !UNITY_EDITOR
-            if (UnityEngine.WSA.Application.RunningOnUIThread())
-#endif
-            {
-                return RequestAccessAndInitAsync(token);
-            }
-#if UNITY_WSA && !UNITY_EDITOR
-            else
-            {
-                UnityEngine.WSA.Application.InvokeOnUIThread(() => RequestAccessAndInitAsync(token), waitUntilDone: true);
-                return Task.CompletedTask;
-            }
+            // FIXME - Use ADM2 instead, this /maybe/ avoids this.
```

Comment:
Did you try with ADM2 if this is still needed? You can forbid access manually from Settings and then run any demo, see if things smoothly error or the app flat out crashes. Last I checked on Desktop with the system-wide authorization (so this is different from the UWP microphone capability we're trying to handle here, though it's the same kind of issue) using ADM2 works around it and will not crash if system-wide microphone access is disabled.

Reply:
It seems to be fixed now, it will raise a few `winrt::hresult_error`s but doesn't seem to crash. I removed this block.